clumio/test/acc_test_lambda.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright 2021. Clumio, Inc.
+#
+import logging
+import json
+import os
+from typing import Any
+
+import boto3
+from botocore import exceptions
+
+
+"""Lambda event handler for acceptance test of clumio_callback_resource."""
+
+
+class Error(Exception):
+    """Base error class."""
+
+
+class PropertyValueError(Error):
+    """A required property is either undefined or empty."""
+
+
+def clumio_event_handler(event: dict[str, Any], _: Any) -> None:
+    """Event handler to process the clumio event."""
+    logger = logging.Logger(name="process_event_logger")
+    try:
+        record = event['Records'][0]
+        resource_properties = json.loads(record['Sns']['Message'])['ResourceProperties']
+    except (KeyError, IndexError, TypeError, json.decoder.JSONDecodeError):
+        raise Error(f'Unable to parse event for ResourceProperties')
+    if type(resource_properties) is not dict:
+        raise Error('ResourceProperties must be of type dictionary.')
+
+    bucket_name = get_required_property(dict(os.environ), 'BUCKET_NAME')
```

Comment:
Instead of creating a new temporary object to send to `get_required_property`, the type hinting could be changed from `dict` to `collections.abc.Mapping` -- which both `dict` and os.environ are derived.